wordpress/wp-content/plugins/memberful-wp/src/contrib/woocommerce.php

```diff
@@ -18,15 +18,27 @@ public static function instance() {
   function __construct() {
     add_action( 'woocommerce_single_product_summary', array( $this, 'hide_add_to_cart_button' ), 25);
     add_filter( 'woocommerce_add_to_cart_validation', array( $this, 'block_cart_add' ), 30, 3 );
-    add_filter( 'woocommerce_is_purchasable', array( $this, 'not_purchasable'), 20, 2 );
+    add_filter( 'woocommerce_is_purchasable', array( $this, 'is_purchasable'), 20, 2 );
+    add_filter( 'the_content', array( $this, 'remove_erroneous_protection' ), -20 );
+  }
+
+  /**
+   * Removes filtering on the content from other functionalities
+   */
+  function remove_erroneous_protection( $content ) {
```

Comment:
@LiamBailey Why do we do this?

Reply:
We do this because when you do add_filter( 'the_content', 'memberful_wp_protect_content', -10 ); the memberful_wp_protect_content catches the_content of all post types. In the case of the WooCommerce integration they want to hide the add to cart button but show the_content which is the product description. So I need to unhook that filter if the post_type is product.

Reply:
Thanks for the explanation!